rotkehlchen/db/calendar.py

```diff
@@ -0,0 +1,219 @@
+                field='name',
+                search_string=name,
+            ))
+        if description is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='description',
+                search_string=description,
+            ))
+        if counterparty is not None:
+            filters.append(DBSubStringFilter(
+                and_op=True,
+                field='counterparty',
+                search_string=counterparty,
+            ))
+        if identifiers is not None:
+            filters.append(DBMultiIntegerFilter(
+                and_op=True,
+                column='identifier',
+                values=identifiers,
+            ))
+
+        filter_query.filters = filters
+        return filter_query
+
+
+class DBCalendar:
+
+    def __init__(self, database: 'DBHandler') -> None:
+        self.db = database
+
+    def create_calendar_entry(
+            self,
+            name: str,
+            timestamp: int,
+            description: str,
+            counterparty: str | None,
+            address: BlockchainAddress | None,
+    ) -> int:
+        """Insert the provided event in the database.
+        May raise:
+        - InputError if any db constraint is not satisfied
+        """
+        with self.db.user_write() as write_cursor:
+            try:
+                write_cursor.execute(
+                    'INSERT INTO calendar(name, timestamp, description, counterparty, address) '
+                    'VALUES (?, ?, ?, ?, ?) RETURNING identifier',
+                    (
+                        name,
+                        timestamp,
+                        description,
+                        counterparty,
+                        address,
+                    ),
+                )
+            except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
+                raise InputError(f'Failed to insert calendar entry {name} due to {e}') from e
+
+            identifier = write_cursor.fetchone()[0]
+        return identifier
+
+    def query_calendar_entry(self, filter_query: CalendarFilterQuery) -> dict[str, Any]:
+        """Query events using the provided filter and return the information required to
+        use pagination filters.
+        """
+        query, bindings = filter_query.prepare()
+        with self.db.conn.read_ctx() as cursor:
+            cursor.execute(
+                'SELECT identifier, name, timestamp, description, counterparty, address FROM '
+                'calendar ' + query,
+                bindings,
+            )
+            result = [CalendarRow(
+                identifier=entry[0],
+                name=entry[1],
+                timestamp=entry[2],
+                description=entry[3],
+                counterparty=entry[4],
+                address=entry[5],
+            ) for entry in cursor]
+
+            entries_total = cursor.execute('SELECT COUNT(*) FROM calendar').fetchone()[0]
```

Comment:
can also use `get_entries_count` right?

Reply:
I can use it but it is basically the same